zookeeper-server/src/test/java/org/apache/zookeeper/test/BackupManagerTest.java

```diff
@@ -526,6 +527,44 @@ public void testTimetableBackup() throws Exception {
     newBackupManager.stop();
   }
 
+  @Test
+  public void testBackupFileNamingOnStreamMode() throws Exception {
+    // Test using a non-default StreamMode to see if the backup file naming is compatible with
+    // different StreamModes used in snapshot compression
+    SnapStream.StreamMode origStreamMode = SnapStream.getStreamMode();
+    SnapStream.setStreamMode(SnapStream.StreamMode.GZIP);
+
+    // Test case 1: make sure backup files are generated correctly with StreamMode extension
+    backupManager.initialize();
+    createNodes(connection, 110);
+    backupManager.getSnapBackup().run(1);
+    String[] expectedBackupFiles = getBackupFiles(backupDir);
+
+    // Sort the snapshot backup file name array because the first snapshot may not have
```

Comment:
Can we do something to make sure the stream mode is set from the beginning?

Reply:
I prefer not setting the stream mode throughout the whole test because that may affect the way other tests are run - they run with the assumption that there's no snapshot compression enabled. 

What we want to test here is whether the backup file names are 1) generated correctly at creation and 2) parsed correctly when read. In order to create an isolated test case, I'm setting a StreamMode only for the scope of this particular test method.